rootfs/scheduler/k8s.py

```diff
@@ -196,8 +270,8 @@ def deploy(self, name, image, command, **kwargs):
             count = 1
             while desired >= count:
                 new_rc = self._scale_app(new_rc_name, count, app_name)
-                if old_rc:
-                    old_rc = self._scale_app(old_rc_name, desired-count, app_name)
+                if old_temp_rc:
```

Comment:
If old_rc scales down to zero then it becomes none and below we are again checking old_rc and deleteting rc which fails.. So using a temp variable while scaling down.